app/src/lib/feature-flag.ts

```diff
@@ -75,3 +75,12 @@ export function enablePullWithRebase(): boolean {
 export function enableGroupRepositoriesByOwner(): boolean {
   return enableBetaFeatures()
 }
+
+/**
+ * As there is some lack of clarity about this feature with how it overlaps
+ * with planned support for stashing changes, this warning dialog is behind
+ * a feature flag for the moment.
+ */
+export function enableLocalChangesWarningHandler(): boolean {
+  return enableDevelopmentFeatures()
```

Comment:
Dev not beta?

Reply:
Yep, here's the PR: https://github.com/desktop/desktop/pull/7016

Reply:
That's correct - the new behaviour has the ability to discard your changes when switching branches, which is the opposite of what we're evaluating with stashed changes, and we want to reconcile both of these approaches so for now we're keeping the old behaviour for `beta` and `production` builds.